core/src/main/java/org/vertexium/FetchHintsBuilder.java

```diff
@@ -55,6 +59,44 @@ public FetchHints build() {
         );
     }
 
+    public FetchHintsBuilder parse(JSONObject fetchHintsJson) {
+        if (fetchHintsJson != null) {
+            this.includeAllProperties = fetchHintsJson.optBoolean("includeAllProperties", false);
+            this.includeAllPropertyMetadata = fetchHintsJson.optBoolean("includeAllPropertyMetadata", false);
+            this.includeHidden = fetchHintsJson.optBoolean("includeHidden", false);
+            this.includeAllEdgeRefs = fetchHintsJson.optBoolean("includeAllEdgeRefs", false);
+            this.includeOutEdgeRefs = fetchHintsJson.optBoolean("includeOutEdgeRefs", false);
+            this.includeInEdgeRefs = fetchHintsJson.optBoolean("includeInEdgeRefs", false);
+            this.includeEdgeLabelsAndCounts = fetchHintsJson.optBoolean("includeEdgeLabelsAndCounts", false);
+            this.includeExtendedDataTableNames = fetchHintsJson.optBoolean("includeExtendedDataTableNames", false);
+
+            if (fetchHintsJson.has("propertyNamesToInclude")) {
+                this.propertyNamesToInclude = jsonArrayToSet(fetchHintsJson.getJSONArray("propertyNamesToInclude"));
+            }
+            if (fetchHintsJson.has("metadataKeysToInclude")) {
+                this.metadataKeysToInclude = jsonArrayToSet(fetchHintsJson.getJSONArray("metadataKeysToInclude"));
+            }
+            if (fetchHintsJson.has("edgeLabelsOfEdgeRefsToInclude")) {
+                this.edgeLabelsOfEdgeRefsToInclude = jsonArrayToSet(fetchHintsJson.getJSONArray("edgeLabelsOfEdgeRefsToInclude"));
+            }
+        }
+
+        return this;
+    }
+
+    private Set<String> jsonArrayToSet(JSONArray jsonArray) {
+        if (jsonArray == null) {
+            return null;
+        }
+
+        List<String> list = new ArrayList<>();
+        for (int i = 0;i < jsonArray.length(); i++) {
+            list.add((String) jsonArray.get(i));
+        }
+
+        return ImmutableSet.<String>builder().addAll(list).build();
```

Comment:
Don't make this an `ImmutableSet` when the builder gets converted to a `FetchHints` it will take care of that